source/common/http/conn_manager_impl.cc

```diff
@@ -1256,7 +1257,8 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ResponseHeaderMap& heade
 
   bool drain_connection_due_to_overload = false;
   if (connection_manager_.drain_state_ == DrainState::NotDraining &&
-      connection_manager_.overload_disable_keepalive_ref_.isSaturated()) {
+      connection_manager_.overload_disable_keepalive_ref_.isRandomizedActive(
+          connection_manager_.random_generator_)) {
```

Comment:
nit: No tests would fail if you reverted the changes to this file. I find that concerning.

Reply:
+1 please add HCM tests.

Reply:
Updated the existing tests so that would fail with the old implementation.